libgadget/blackhole.c

```diff
@@ -1396,6 +1402,24 @@ blackhole_feedback_reduce(int place, TreeWalkResultBHFeedback * remote, enum Tre
     TREEWALK_REDUCE(BHP(place).CountProgs, remote->BH_CountProgs);
 }
 
+/* Sample from a power law to get the initial BH mass*/
+static double
+bh_powerlaw_seed_mass(MyIDType ID)
+{
+    /* compute random number, uniform in [0,1] */
+    const double w = get_random_number(ID);
```

Comment:
Perhaps add an offset to ID to decorrelate with other random numbers seeded on the particle.